function-sidecar/pkg/dispatcher/grpc/grpc.go

```diff
@@ -52,59 +58,157 @@ func (this *grpcDispatcher) Closed() <-chan struct{} {
 	return this.closed
 }
 
-func (this *grpcDispatcher) handleIncoming() {
+// A CorrelationStrategy groups together messages that belong to the same "spatial" stream (the "temporal" aspect
+// of grouping is taken care of by a WindowStrategy). Several invocations for different spatial groups may happen concurrently.
+type CorrelationStrategy func(message.Message) interface{}
+
+// A WindowingStrategyFactory is a function for creating a new instance of a (most certainly stateful) WindowingStrategy,
+// given a new, first message for a stream.
+type WindowingStrategyFactory func(message.Message) WindowingStrategy
+
+//
+type WindowingStrategy interface {
+	// ShouldClose notifies this strategy that a new message has just been sent to the function, resulting in the optional
+	// error passed as 2nd argument. This is a synchronous way of deciding whether or not a window should be closed.
+	// ShouldClose should return true if this strategy decides that the current stream should end.
+	ShouldClose(in message.Message, err error) bool
+
+	// AsyncClose should return a non-nil channel that this strategy will close if it decides that the current window
+	// should end asynchronously
```

Comment:
I'm being pedantic about the English, but shouldn't the comment say "... if it decides asynchronously that the current window should end."? It's like an async equivalent to `ShouldClose` returning `true` isn't it?

Reply:
Absolutely, will change